tests/framework/integration.go

```diff
@@ -34,9 +36,40 @@ func (e integrationRunner) BeforeTest(t testing.TB) {
 	integration.BeforeTest(t)
 }
 
-func (e integrationRunner) NewCluster(t testing.TB) Cluster {
+func (e integrationRunner) NewCluster(t testing.TB, cfg config.ClusterConfig) Cluster {
+	integrationCfg := integration.ClusterConfig{
+		Size: cfg.ClusterSize,
+	}
+	switch cfg.ClientTLS {
+	case config.NoTLS:
+		integrationCfg.ClientTLS = nil
+	case config.ManualTLS:
+		integrationCfg.ClientTLS = &integration.TestTLSInfo
+	case config.AutoTLS:
+		tls, err := transport.SelfCert(zap.NewNop(), t.TempDir(), []string{"localhost"}, 1)
+		if err != nil {
+			t.Fatalf("Failed to generate ClientTLS cert: %s", err)
+		}
+		integrationCfg.ClientTLS = &tls
+	default:
+		t.Fatalf("ClientTLS config %q not supported", cfg.ClientTLS)
+	}
+	switch cfg.PeerTLS {
+	case config.NoTLS:
+		integrationCfg.PeerTLS = nil
+	case config.AutoTLS:
+		tls, err := transport.SelfCert(zap.NewNop(), t.TempDir(), []string{"localhost"}, 1)
+		if err != nil {
+			t.Fatalf("Failed to generate PeerTLS cert: %s", err)
+		}
+		integrationCfg.PeerTLS = &tls
+	case config.ManualTLS:
+		integrationCfg.PeerTLS = &integration.TestTLSInfo
+	default:
+		t.Fatalf("PeerTLS config %q not supported", cfg.PeerTLS)
+	}
```

Comment:
The ClientTLS and PeerTLS follow the same logic, so please try to create a generic function something like `generateTLSInfo` to generate the TLSInfo based on the provided configuration. Afterwards, the above code can be simplified as below,

```
integrationCfg.ClientTLS = generateTLSInfo(t, cfg.ClientTLS)
integrationCfg.PeerTLS = generateTLSInfo(t, cfg.PeerTLS)
```